README.md

```diff
@@ -2,27 +2,170 @@
 
-- FluentBit: Defines Fluent Bit instances and associated ConfigMap
-- FluentBitConfig: Select Input/Filter/Output and generates ConfigMap
-- Input: Defines tail plugins
-- Filter: Defines nest/modify/kubernetes plugins 
-- Output: Defines forward/kafka/es/stdout plugins
+```shell
+kubectl apply -f manifests/setup
+kubectl apply -f manifests/logging-stack
+```
 
-## Features
+Within a couple of minutes, you should observe an index available:
+
+```shell
+$ curl localhost:9200/_cat/indices
+green open ks-logstash-log-2020.04.26 uwQuoO90TwyigqYRW7MDYQ 1 1  99937 0  31.2mb  31.2mb
+``` 
+
+Success!
+
+## API Doc
+
+The listing below shows supported plugins currently. It is based on Fluent Bit v1.3.7. For more information, see API docs of each plugin.
+
+- [Input](docs/crd.md#input)
+    - [dummy](docs/plugins/input/dummy.md)
+    - [tail](docs/plugins/input/tail.md)
+- [Filter](docs/crd.md#filter)
+    - [kubernetes](docs/plugins/filter/kubernetes.md)
+    - [modify](docs/plugins/filter/modify.md)
+    - [nest](docs/plugins/filter/nest.md) 
+- [Output](docs/crd.md#output)
+    - [elasticsearch](docs/plugins/output/elasticsearch.md)
+    - [forward](docs/plugins/output/forward.md)
+    - [kafka](docs/plugins/output/kafka.md)
+    - [null](docs/plugins/output/null.md)
+    - [stdout](docs/plugins/output/stdout.md)
+
+## Best Practice
+
+### Plugin Grouping
+
+Input, filter and output plugins are connected by the mechanism of tagging and matching. For input and output plugins, always create `Input` or `Output` instances for every plugin. Don't aggregate multiple inputs or outputs into one `Input` or `Output` object, except you have a good reason to do so. Take the demo logging stack for example, we have independent yaml files for each kind of outputs.
+
+However, for filter plugins, if you want a filter chain, order of filters matters. You need organize multiple filters into an array as the demo [logging stack](manifests/logging-stack/filter-kubernetes.yaml) suggests.
+
+### Path Convention
+
+Path to file in Fluent Bit config should be well regulated. Fluent Bit Operator adopts the following convention internally.
+
+|Dir Path|Description|
+|---|---|
+|/fluent-bit/tail|Stores tail related files, eg. file tracking db. Using [fluentbit.spec.positionDB](docs/crd.md#fluentbitspec) will mount a file `pos.db` under this dir by default.|
+|/fluent-bit/secrets/{secret_name}|Stores secrets, eg. TLS files. Specify secrets to mount in [fluentbit.spec.secrets](docs/crd.md#fluentbitspec), then you have access.|
+|/fluent-bit/config|Stores the final config file.|
+
+> Note that ServiceAccount files are mounted at `/var/run/secrets/kubernetes.io/serviceaccount`.
+
+## Features In Plan
 
```

Comment:
3 aws fluentbit plugin should be in the plan:
amazon-kinesis-firehose-for-fluent-bit: https://github.com/aws/amazon-kinesis-firehose-for-fluent-bit
amazon-cloudwatch-logs-for-fluent-bit: https://github.com/aws/amazon-cloudwatch-logs-for-fluent-bit
amazon-kinesis-streams-for-fluent-bit: https://github.com/aws/amazon-kinesis-streams-for-fluent-bit

these plugins was provided in amazon/aws-for-fluent-bit container image: https://github.com/aws/aws-for-fluent-bit, we could investigate if it's possible to integrate these 3 plugins without using aws fluentbit image

Reply:
Update readmap: Support custom Input/Filter/Output plugins